### PR TITLE
Test server

### LINKED ISSUE #3: Implement Test HTTPS Server
- [x] Implement test server that provides [Lemon API](https://github.com/AlmostGreatBand/lemon-docs) and runs on localhost. Server can be launched by `npm run testServer` command.

Optional: write tests for it.

### PR DESCRIPTION
**Pull request by MaryLynJuana**
Implement test server that provides Lemon API and runs on localhost. 
Server can be launched by npm run testServer command.
Closes #3

### CODE REVIEW

File: src/testServer.js [hunk 1]
```diff
@@ -1,5 +1,206 @@
...
+    date: '2020-10-13T15:37:01.325Z'
+  },
+  {
+    id: 3,
+    card: 657,
+    amount: -1236,
+    type: 'Rent',
+    date: '2020-10-18T14:10:23.753Z'
+  }],
+};
```
Comment (ALEGATOR1209) [inline src/testServer.js]: Maybe it'd be a better idea to put this data in some external fle, idk? 

Like sertificates above.

File: src/testServer.js [hunk 2]
```diff
@@ -1,5 +1,206 @@
...
+
+const getBanks = accId => {
+  const banks = [];
+  for (const bank of db.banks) {
+    if (bank.account === accId) banks.push(bank);
+  }
+  return banks;
+};
+const getCards = accId => {
```
Comment (ALEGATOR1209) [inline src/testServer.js]: Add blank line)0)00)00)))

**Review by ALEGATOR1209 — CHANGES_REQUESTED** [review]
Waiting till testing system will be ready.

**Review by Hud-Volodymyr — COMMENTED** [review]
Put db, handlers, data access and routing into different files(e.g. db.test.js, controllers.test.js, etc.)

File: src/testServer.js [hunk 3]
```diff
@@ -1,5 +1,206 @@
...
+  return transactions;
+};
+
+const routing = {
+  '/': () => '<h1>Welcome to Lemon&#127819 Server!</h1>',
+  '/favicon.ico/': () => {}, //Debug, never mind
+  '/profile/': (req, res) => profileHandler(req, res),
+  '/cards/': (req, res) => cardsHandler(req, res),
+  '/transactions/': (req, res) => transactionsHandler(req, res),
+  '/registration': () => '<h1>&#127819Registration process will be here</h1>',
```
Comment (Hud-Volodymyr) [inline src/testServer.js]: Since we have normilizePath function which adds '/' to every route '/registration' will not be accessible

**Review by Hud-Volodymyr — CHANGES_REQUESTED** [review]

File: src/dbInterface.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+};
+
+const getCards = accId => {
+  const banks = getBanks(accId);
+  const cards = [];
+  for (const card of db.cards) {
+    for (const bank of banks) {
+      if (card.bank === bank.id) cards.push(card);
+    }
+  }
```
Comment (ALEGATOR1209) [inline src/dbInterface.js]: Maybe more functional style?

```suggestion
  const banks = getBanks(accId).map(bank => bank.id);
  return db.cards.filter(card => banks.includes(card.bank));
```

File: src/dbInterface.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  const transactions = [];
+  for (const transaction of db.transactions) {
+    for (const card of cards) {
+      if (transaction.card === card.id) {
+        transactions.push(transaction);
+      }
+    }
+  }
+  return transactions;
+};
```
Comment (ALEGATOR1209) [inline src/dbInterface.js]: Can be simplified to functional style too, I think

File: src/routing.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  if (password !== account.password) return {
+    err: '<h1>&#127819Incorrect password entered :(</h1>',
+    account: null
+  };
+  return { err: null, account };
+};
+
+const profileHandler = (req, res) => {
+  const login = url.parse(req.url, true).query.login; //const login = req.getHeader('login');
+  const password = url.parse(req.url, true).query.password; //const password = req.getHeader('password');
```
Comment (ALEGATOR1209) [inline src/routing.js]: Isn't `url.parse` deprecated?

**Review by ALEGATOR1209 — COMMENTED** [review]
P. S. On my mind, it'd be better to move it to src/testServer folder.

File: src/routing.js (hunk 6 above)
Comment (Hud-Volodymyr) [reply src/routing.js]: It's indeed deprecated, use URL constructor instead.

**Review by Hud-Volodymyr — DISMISSED** [review]

File: package.json [hunk 7]
```diff
@@ -15,5 +15,9 @@
   "keywords": [
     "lemon"
   ],
-  "author": "AlmostGreatBand"
+  "author": "AlmostGreatBand",
+  "dependencies": {
+    "eslint": "^7.11.0",
+    "vue": "^2.6.12"
```
Comment (ALEGATOR1209) [inline package.json]: Vue on server side?)))

Also, ESLint will be added by #8 when we merge it to master.

**Review by ALEGATOR1209 — APPROVED** [review]

**Review by Hud-Volodymyr — APPROVED** [review]

**Pull request merged**